arangod/IResearch/IResearchLink.cpp

```diff
@@ -321,18 +324,39 @@ Result IResearchLink::init(velocypack::Slice definition,
         }
 #endif
       }
+    }
 
-      if (!clusterWideLink) {
-        // prepare data-store which can then update options
-        // via the IResearchView::link(...) call
-        auto res = initDataStore(initCallback, meta._version, sorted,
-                                 storedValuesColumns, primarySortCompression);
+    if (!clusterWideLink) {
+      if (meta._collectionName.empty() && !clusterIsEnabled &&
```

Comment:
we can reduce risk even more in case if we can check if `_id` attribute is indexed.